lib/runtime/MakeDeferredNamespaceObjectRuntime.js

```diff
@@ -0,0 +1,212 @@
+	"// optimization not applied when output.strictModuleErrorHandling is off"
+];
+
+class MakeDeferredNamespaceObjectRuntimeModule extends HelperRuntimeModule {
+	/**
+	 * @param {boolean} hasAsyncRuntime if async module is used.
+	 */
+	constructor(hasAsyncRuntime) {
+		super("make deferred namespace object");
+		this.hasAsyncRuntime = hasAsyncRuntime;
+	}
+
+	/**
+	 * @returns {string | null} runtime code
+	 */
+	generate() {
+		const { runtimeTemplate } = this.compilation;
+		const fn = RuntimeGlobals.makeDeferredNamespaceObject;
+		const hasAsync = this.hasAsyncRuntime;
+		const strictError =
+			this.compilation.options.output.strictModuleErrorHandling;
+		return `${fn} = ${runtimeTemplate.basicFunction("moduleId, mode", [
+			"// mode: 0 => namespace (esm)",
+			"// mode: 1 => default-only (esm strict cjs)",
+			"// mode: 2 => default-with-named (esm-cjs compat)",
+			"// mode: 3 => dynamic (if exports has __esModule, then esm, otherwise default-with-named)",
+			"",
+			"var cachedModule = __webpack_module_cache__[moduleId];",
+			...(strictError ? strictModuleCache : nonStrictModuleCache),
+			"",
+			"if (mode == 1) {",
+			Template.indent([
+				// if this var is changed to let or const, please define a new ns in the outer scope.
+				"var ns = { __proto__: null };",
+				`${RuntimeGlobals.makeNamespaceObject}(ns);`,
+				`${RuntimeGlobals.definePropertyGetters}(ns, {`,
+				Template.indent([
+					'"default": ' +
+						runtimeTemplate.returningFunction(
+							`${RuntimeGlobals.require}(moduleId)`
+						)
+				]),
+				"});",
+				"return ns;"
+			]),
+			"}",
+			"",
+			`var init = ${runtimeTemplate.basicFunction("", [
+				`ns = ${RuntimeGlobals.require}(moduleId);`,
+				hasAsync
+					? `if (${RuntimeGlobals.asyncModuleExportSymbol} in ns) ns = ns[${RuntimeGlobals.asyncModuleExportSymbol}];`
+					: "",
+				"init = null;",
+				"if (mode == 3) {",
+				Template.indent(["if (ns.__esModule) mode = 0;", "else mode = 2;"]),
+				"}",
+				"if (mode) return ns;",
+				// this optimization only applies to esm because it's .exports cannot be assigned to another object.
+				"delete handler.defineProperty;",
+				"delete handler.deleteProperty;",
+				"delete handler.set;",
+				"delete handler.get;",
+				"delete handler.has;",
+				"delete handler.ownKeys;",
+				"delete handler.getOwnPropertyDescriptor;",
+				"return ns;"
+			])};`,
+			"",
+			`ns = ${
+				strictError ? "" : "cachedModule && cachedModule.exports || "
+			}__webpack_module_deferred_exports__[moduleId] || (__webpack_module_deferred_exports__[moduleId] = { __proto__: null });`,
+			"var handler = {",
+			Template.indent([
+				"__proto__: null,",
+				`get: ${runtimeTemplate.basicFunction("_, name", [
+					"switch (name) {",
+					Template.indent([
+						'case "__esModule": return true;',
+						'case Symbol.toStringTag: return "Module";'
+					]),
+					"}",
+					runtimeTemplate.supportsOptionalChaining
+						? "init?.();"
+						: "if (init) init();",
```

Comment:
The new tests are failing on Node.js 10 because the bundle contains `?.`, which is not supported there. What's the best way to change the tests in this case?